// This is used for convolution logging. Also see
// tensorflow/core/protobuf/autotuing.h
syntax = "proto3";

package tensorflow;

import "tensorflow/stream_executor/dnn.proto";

// A convolution. Currently it's only used for logging. In the future, we may
// want to use it in the API as well.
message ConvolutionProto {
  stream_executor.dnn.ConvolutionKind kind = 1;
  stream_executor.dnn.TensorDescriptorProto input = 2;
  stream_executor.dnn.TensorDescriptorProto filter = 3;
  stream_executor.dnn.TensorDescriptorProto output = 4;
  stream_executor.dnn.ConvolutionDescriptorProto conv_desc = 5;

  // result = conv_scale * conv(...) + side_value_scale * side_value.
  // side_value is an arbitrary buffer if activation is not none. Otherwise, it
  // has to be the result buffer (using its old values).
  double conv_scale = 6;
  double side_value_scale = 7;

  stream_executor.dnn.ActivationMode activation = 8;
}
